Modules/Loadable/SubjectHierarchy/Widgets/qSlicerSubjectHierarchyAbstractPlugin.cxx

```diff
@@ -284,6 +284,31 @@ QString qSlicerSubjectHierarchyAbstractPlugin::tooltip(vtkIdType itemID) const
   return tooltipString;
 }
 
+//-----------------------------------------------------------------------------
+QString qSlicerSubjectHierarchyAbstractPlugin::tooltipWithStoragePath(vtkIdType itemID) const
+{
+  vtkMRMLSubjectHierarchyNode* shNode = qSlicerSubjectHierarchyPluginHandler::instance()->subjectHierarchyNode();
+  if (!shNode)
+  {
+    qCritical() << Q_FUNC_INFO << ": Failed to access subject hierarchy node";
+    return QString();
+  }
+
+  // If associated data node is a storable node, get the file path from the URL
+  vtkMRMLNode* dataNode = shNode->GetItemDataNode(itemID);
+  if (dataNode != nullptr)
+  {
+    vtkMRMLStorableNode* storableNode = vtkMRMLStorableNode::SafeDownCast(dataNode);
+    if (storableNode != nullptr && storableNode->GetStorageNode() != nullptr)
+    {
+      QString tooltipWithStoragePath = tr("Loaded from file: ");
+      tooltipWithStoragePath.append(storableNode->GetStorageNode()->GetFileName());
```

Comment:
GetFileName() may return nullptr causing crash, so we need a null-pointer check.
Also, we probably don't want to show the line if the string is empty.